chart/tests/test_elasticsearch_secret.py

```diff
@@ -96,3 +96,24 @@ def test_should_correctly_handle_password_with_special_characters(self):
             "http://username%21%40%23$%25%25%5E&%2A%28%29:password%21%40%23$%25%25%5E&%2A%28%29@"
             "elastichostname:80" == connection
         )
+
+    def test_should_generate_secret_with_specified_port(self):
+        connection = self._get_connection(
+            {
+                "elasticsearch": {
+                    "enabled": True,
+                    "connection": {
+                        "user": "username!@#$%%^&*()",
+                        "pass": "password!@#$%%^&*()",
+                        "host": "elastichostname",
+                        "port": 9200,
+                    },
+                }
+            }
+        )
+
+        assert (
+            "http://username%21%40%23$%25%25%5E&%2A%28%29:password%21%40%23$%25%25%5E&%2A%28%29@"
+            "elastichostname:9200" == connection
+        )
```

Comment:
```suggestion
                        "host": "elastichostname",
                        "port": 2222,
                    },
                }
            }
        )

        assert "http://elastichostname:2222" == connection
```

Let's keep it simple and also use a non-default port?

Reply:
Good. I committed it.

Reply:
OK. It turns out it is not good. 
Without user and pass field, `<no+value>` is included in url string, like this.
`http://<no+value>:<no+value>@elastichostname:2222`

It makes test fail.

![image](https://user-images.githubusercontent.com/26663842/128814721-114465cc-92aa-4b02-9c58-195c33d3aeed.png)

My mistake, I should've tested for this.

Reply:
Ugh, no worries, just put in simple values for now. Might be another follow up we need to look at.

Reply:
Resolve:
https://github.com/apache/airflow/pull/17428/commits/ef2e80b6fe6c747b040ea7f4954f8ae95a7aefa1